src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -381,3 +381,82 @@ def linear_isometry_equiv.from_orthogonal_span_singleton
   (n : ℕ) [fact (finrank 𝕜 E = n + 1)] {v : E} (hv : v ≠ 0) :
   (𝕜 ∙ v)ᗮ ≃ₗᵢ[𝕜] (euclidean_space 𝕜 (fin n)) :=
 linear_isometry_equiv.of_inner_product_space (finrank_orthogonal_span_singleton hv)
+
+section complex
+
+variables {V : Type*}
+[inner_product_space ℂ V]
+[finite_dimensional ℂ V]
+
+variables {S : submodule ℂ V} {L : S →ₗᵢ[ℂ] V}
+
+open finite_dimensional
+
+/-- Let `S` be a subspace of a finite-dimensional complex inner product space `V`.  A linear
+isometry mapping `S` into `V` can be extended to a full isometry of `V`. -/
+noncomputable def linear_isometry.extend (L : S →ₗᵢ[ℂ] V): V →ₗᵢ[ℂ] V :=
+begin
+  -- Build an isometry from Sᗮ to L(S)ᗮ through euclidean_space
+  let d := finrank ℂ Sᗮ,
+  have dim_S : finrank ℂ Sᗮ = d := rfl,
+  let LS := L.to_linear_map.range,
+  have dim_LS_perp : finrank ℂ LSᗮ = d,
+  calc  finrank ℂ LSᗮ = finrank ℂ V - finrank ℂ LS : by simp only
+      [← LS.finrank_add_finrank_orthogonal, add_tsub_cancel_left]
+    ...               = finrank ℂ V - finrank ℂ S : by simp only
+      [linear_map.finrank_range_of_inj L.injective]
+    ...               = finrank ℂ Sᗮ : by simp only
+      [← S.finrank_add_finrank_orthogonal, add_tsub_cancel_left]
+    ...               = d : dim_S,
+  let L1 := ((fin_std_orthonormal_basis dim_S).to_orthonormal_basis
+    (fin_std_orthonormal_basis_orthonormal dim_S)).repr.to_linear_isometry,
+  let L2 := ((fin_std_orthonormal_basis dim_LS_perp).to_orthonormal_basis
+    (fin_std_orthonormal_basis_orthonormal dim_LS_perp)).repr.symm.to_linear_isometry,
```

Comment:
Can you extract a separate definition that takes in the `orthonormal_basis` as an argument, rather than choosing it nonuniquely here?

Reply:
Which `orthonormal_basis` would you expect as the argument?  One for the subspace `S`?

Reply:
Looking again, we're deep into noncomputable territory anyway, so I don't think this is worth attempting.